lib/dal/src/schema/variant.rs

```diff
@@ -1084,6 +1084,27 @@ impl SchemaVariant {
         Ok(Func::get_by_id_or_error(ctx, asset_func_id).await?)
     }
 
+    /// This method unlocks the asset [`Func`] without creating a copy.
+    ///
+    /// **Warning:** this is a somewhat dangerous as we should normally create a copy of an asset
```

Comment:
Why is it dangerous?

Reply:
Anytime we unlock, we want to copy the object being unlocked. This specific issue that the PR tackles is a "special case".